packages/lib/services/search/SearchEngine.ts

```
@@ -777,18 +777,29 @@ export default class SearchEngine {
 				if (!queryHasFilters) {
 					const toSearch = parsedQuery.allTerms.map(t => t.value).join(' ');
 
-					let itemRows = await this.db().selectAll<ProcessResultsRow>(`
-						SELECT
-							id,
-							title,
-							user_updated_time,
-							offsets(items_fts) AS offsets,
-							matchinfo(items_fts, 'pcnalx') AS matchinfo,
-							item_id,
-							item_type
-						FROM items_fts
-						WHERE title MATCH ? OR body MATCH ?
-					`, [toSearch, toSearch]);
+					let itemRows: ProcessResultsRow[] = [];
+
+					try {
+						itemRows = await this.db().selectAll<ProcessResultsRow>(`
+								SELECT
+									id,
+									title,
+									user_updated_time,
+									offsets(items_fts) AS offsets,
+									matchinfo(items_fts, 'pcnalx') AS matchinfo,
+									item_id,
+									item_type
+								FROM items_fts
+								WHERE title MATCH ? OR body MATCH ?
+						`, [toSearch, toSearch]);
+					} catch (error) {
```

What specific error is thrown for Android <= 25? I think we should check and do nothing only for that particular error. Otherwise, we should rethrow the error

---

Here's an example of the error:
```
Error Code : 1 (SQLITE_ERROR)
Caused By : SQL(query) error or missing database.
        (unable to use function MATCH in the requested context (code 1))
```
As it doesn't contain the text `syntax error`, we can't use `isSqliteSyntaxError` as is done in the other location with Android API <= 25 compatibility code.

I've added logic to re-throw the exception if its message doesn't contain `unable to use function MATCH in the requested context`.